fe/fe-core/src/main/java/com/starrocks/connector/iceberg/rest/IcebergRESTCatalog.java

```diff
@@ -123,9 +131,26 @@ public boolean tableExists(String dbName, String tableName) throws StarRocksConn
 
     @Override
     public List<String> listAllDatabases() {
-        return delegate.listNamespaces().stream()
-                .map(ns -> ns.level(0))
-                .collect(Collectors.toList());
+        if (nestedNamespaceEnabled) {
```

Comment:
I think there is a better way of writing
```
try {
 if(nest) {
  ...
 } else {
..
 }
} catch(REST) {
}```

So you need to catche xception only once.